examples/pln/johnny/Assertions.scm

```diff
@@ -0,0 +1,50 @@
+(ConceptNode "Johnny@444") (stv 0.99 1)
+
+(ConceptNode "Sweets@555") (stv 0.99 1)
+
+(ConceptNode "day@222") (stv 0.99 1)
+
+(ConceptNode "he@345") (stv 0.99 1)
+
+(ConceptNode "himself@456") (stv 0.99 1)
+
+(ConceptNode "risk@44") (stv 0.99 1)
+
+(ConceptNode "diabetes@66") (stv 0.99 1)
+
+(ImplicationLink (stv 0.99 1)
+	(EvaluationLink
+		(PredicateNode "every@111")
+        (ListLink
+			(ConceptNode "day@222")
+			(EvaluationLink
+				(PredicateNode "eating@333")
+				(ListLink
+					(ConceptNode "Johnny@444")
+					(ConceptNode "Sweets@555")))))
+	(EvaluationLink
+		(PredicateNode "at@123")
+		(ListLink
+			(EvaluationLink
+				(PredicateNode "placing@234")
+				(ListLink
+					(ConceptNode "he@345")
+					(ConceptNode "himself@456")))
+			(ConceptNode "risk@44"))))
+
+(EvaluationLink (stv 0.99 1)
+	(PredicateNode "for@55")
+	(ListLink
+		(ConceptNode "risk@44")
+		(ConceptNode "diabetes@66")))
+
+(NotLink (stv 0.99 1)
+	(EvaluationLink
+		(PredicateNode "every@111")
+        (ListLink
+			(ConceptNode "day@222")
+			(EvaluationLink
+				(PredicateNode "eating@333")
+				(ListLink
+					(ConceptNode "Johnny@444")
+					(ConceptNode "Sweets@555"))))))
```

Comment:
> . . .  It's like trying
> to put together one of those puzzles made of wooden blocks, where as soon
> as you fit one piece in, three of the ones
> that were in before, fall out . . .
>
>
Well, that is exactly the optimization problem I keep talking about. You
cannot just add on new rules, without also going back to modify the rules
you learned previously.  The older rules were "correct" when dealing with
the smaller corpus, but, when enlarging your corpus, you find that some of
the old rules result in bad outcomes, bad deductions.

You have to go back and modify the old rules.  This is like Ben's email
where he asks "what N can I choose for language" and the answer is you
can't: its neither global nor static. You worked with a small value of N at
first, and learned some confident rules for a limited corpus.

As you start hearing, handling, processing new classes of sentences, one of
two things happen: either some of the existing rules are confirmed, in
which case your confidence of those rules get *even bigger*, their status
as bedrock rules is further strengthened.  The other possibility is that
the existing rules get modified.  To find the right modification, you have
to go to deeper bedrock, and solve the global problem of (deeper_bedrock +
modified_rules + new_corpus) is minimally conflicted, inconsistent.

This makes the learning process hierarchical, and buries some rules deeply,
where they can't be modified without tearing down the whole edifice. Other
rules remain superficial and malleable, and don't harden into a fixed form
until a lot of corpus experience is built up.

-- Linas